### DISCUSSION TITLE
How to measure the pupil measurement detection accuracy

### DISCUSSION THREAD

**Discussion by QuinzyW** [openPupil/Open-PupilEXT#24] [General]
hello
On paper, the author represents a five-pixel error, and at the start of the paper .`A developed 120-fps pupillometry demo system was able to achieve a calibration accuracy of 0.003 mm and an averaged temporal pupil measurement detection accuracy of 0.0059 mm in stereo mode. `. So what's the distance between camera and subject or how to convert the five-pixel error to an accurate 0.003mm error.

**Comment by BZandi** [comment]
Verifying the pupil measurement accuracy in absolute units (mm) can only be done when operating PupilEXT in stereo mode (two cameras), as only with two cameras the pupil size can be measured in millimetres. Therefore, the two values that you reported are from our stereo camera demo setup.

When using a single camera, the pupil diameter is given in pixels, meaning converting the value to millimetres needs to be done manually when pre-processing the data. For this, a circular object of known size can be placed in front of the subject's eye (after the experiment) to obtain the transformation scale from pixel to mm (see Figure). The circular reference object can be recorded in PupilEXT, with which you can derive a relationship between Pixel size and the known size of the reference object. However, when using such a conversion technique, the distance from the subjects' eye to the camera must remain steady during the pupil measurement and while recording the reference object, which can be achieved using a chin rest, for example.

![Reference_object](https://user-images.githubusercontent.com/62992112/161019572-a04e1e80-df71-4b0a-9ab9-cf4d56ecc168.png)

For single-camera setups, it is recommended to report the camera's optical resolution, the camera-to-eye distance and the transformation constant used to convert the pixel values into mm (retrieved from the reference object measurement). Additionally, the intrinsic camera calibration results must be reported, as stated in PupilEXT.
